[Lim Juntong] iP

lim-jt opened the pull request:
# Dash

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Dash frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is,

1. download it from [here](https://github.com/lim-jt/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

 - [X] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)


If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rename main class and add greeting

Commit:
Revert "Rename main class and add greeting"

This reverts commit 4527a4e6cbb8c9d6bdbc7ae231493aedb4427931.

Commit:
Rename and add greeting

Commit:
Rename .java file

Commit:
Add ASCII art

Commit:
Add ASCII goodbye and echo loop

Commit:
Add ability to add and list text

Commit:
Add new class Task, added mark and unmark features

Commit:
changed toString() for Task and added todo deadline event tasks

Commit:
text ui test

Commit:
Added exceptions

Commit:
Added delete feature

lim-jt committed:
Added save feature

lim-jt committed:
Revert "Added save feature"

This reverts commit 98abf22f252d6d565e67ccfed00b01135f6c5d06.

lim-jt committed:
Revert "Revert "Added save feature""

This reverts commit 412548eb7f575de37c9eac51e176537cbf19c654.

lim-jt committed:
Merge branch 'branch-Level-7'

lim-jt committed:
Add date and time recognition feature

lim-jt committed:
Revert "Add date and time recognition feature"

This reverts commit 1947a13fabcbee0890e2fb3b681bd6254f5317d2.

lim-jt committed:
Revert "Revert "Add date and time recognition feature""

This reverts commit 7c3263154053e2714180362e45c33fc970235440.

lim-jt committed:
Merge branch 'brance-Level-8'

lim-jt committed:
Add DashException that encapsulates all Dash-related exceptions

SSirMentos commented on src/main/java/Dash.java:
```diff
@@ -0,0 +1,305 @@
+                        System.out.println(HORIZONTAL_LINE);
+                        System.out.println("Nice! I've marked this task as done:");
+                        System.out.println(t);
+                        System.out.println(HORIZONTAL_LINE);
```
Repeated code of printing, maybe can use a UI class to handle all of the printing

SSirMentos commented on src/main/java/Dash.java:
```diff
@@ -0,0 +1,305 @@
+                    } else if (type == 'D') {
+                        Deadline d = new Deadline(stringArray[2], LocalDateTime.parse(stringArray[3], fileFormatter));
+                        if (stringArray[1].equals("X")) {
+                            d.markTaskAsDone();
```
repeated code. Maybe can put in a function that accept type Task

SSirMentos commented on src/main/java/Dash.java:
```diff
@@ -0,0 +1,305 @@
+                        throw new IncorrectCommandUseException("Please include the /to command.");
+                    }
+
+                    LocalDateTime from = LocalDateTime.parse(fromString[1], inputFormatter);
```
Repeated code of parsing date, perhaps create a class to handle these parsing?

SSirMentos commented on src/main/java/Dash.java:
```diff
@@ -0,0 +1,305 @@
+
+        //Exit
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println(GOODBYE);
```
Perhaps create a UI class that has a goodbye function to print

lim-jt committed:
Add OOP features, add command class

lim-jt committed:
Merge branch 'branch-A-MoreOOP'

lim-jt committed:
Add packages

lim-jt committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

lim-jt committed:
Add gradle support

lim-jt committed:
Add JUnit test cases for TaskList and Todo

lim-jt committed:
Add executable jar file

lim-jt committed:
Add find command

lim-jt committed:
Add javadoc and coding standard changes

lim-jt committed:
Merge branch 'branch-A-CodingStandard'

lim-jt committed:
Merge branch 'master' into branch-Level-9

lim-jt committed:
Merge branch 'branch-Level-9' into branch-A-JavaDoc

yimjianbing commented on src/main/java/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,24 @@
+import java.time.LocalDateTime;
+
+public class DeadlineCommand extends Command {
+    private final String description;
+    private final LocalDateTime deadline;
```
your variable name supposed to be in all caps la bodoh, its a constant!!! never read coding standard tsk..

lim-jt replied:
bro static final then is constant lah stupid, i still need to initialise this variable in the constructor

yimjianbing commented on src/main/java/main/Parser.java:
```diff
@@ -0,0 +1,135 @@
+package main;
+
+import command.*;
```
u puborr kia!! u can't anyhow import using .*!!! you need to specify which class u want if not later prof angy

lim-jt replied:
valid... intellij auto add one

yimjianbing commented on src/main/java/main/Parser.java:
```diff
@@ -0,0 +1,135 @@
+ * Handles command parsing and validation, and manages exceptions.
+ */
+public class Parser {
+    private static final DateTimeFormatter INPUT_FORMATTER = DateTimeFormatter.ofPattern("d-M-yyyy HHmm");
```
wow u actly named it in all caps u smarty pants <33 good job bb

yimjianbing commented on src/main/java/main/Storage.java:
```diff
@@ -0,0 +1,185 @@
+                }
+                System.out.println("Existing data file found. Data has been loaded.");
+            } else {
+                if (file.getParentFile().exists()) {
+                    file.createNewFile();
+                } else {
+                    file.getParentFile().mkdir();
+                    file.createNewFile();
+                }
+                System.out.println("No existing data file found. New data file \"dash.txt\" has been created.");
```
confirm is chatgpt do for u one, dont lie i dont believe u so smart

lim-jt replied:
bruh i legit ownself did this u think chatgpt so smart??

yimjianbing commented on src/main/java/main/Ui.java:
```diff
@@ -0,0 +1,195 @@
+        this.scanner = new Scanner(System.in);
+    }
+
+    private static final String LOGO = " .----------------.  .----------------.  .----------------.  .----------------.\n" +
+            "| .--------------. || .--------------. || .--------------. || .--------------. |\n" +
+            "| |  ________    | || |      __      | || |    _______   | || |  ____  ____  | |\n" +
+            "| | |_   ___ `.  | || |     /  \\     | || |   /  ___  |  | || | |_   ||   _| | |\n" +
+            "| |   | |   `. \\ | || |    / /\\ \\    | || |  |  (__ \\_|  | || |   | |__| |   | |\n" +
+            "| |   | |    | | | || |   / ____ \\   | || |   '.___`-.   | || |   |  __  |   | |\n" +
+            "| |  _| |___.' / | || | _/ /    \\ \\_ | || |  |`\\____) |  | || |  _| |  | |_  | |\n" +
+            "| | |________.'  | || ||____|  |____|| || |  |_______.'  | || | |____||____| | |\n" +
+            "| |              | || |              | || |              | || |              | |\n" +
+            "| '--------------' || '--------------' || '--------------' || '--------------' |\n" +
+            " '----------------'  '----------------'  '----------------'  '----------------'";
```
cringe.

lim-jt replied:
u just jealous

yimjianbing reviewed (commented):
L code ngl

lim-jt committed:
Update README.md

lim-jt committed:
Merge branch 'branch-A-JavaDoc'

lim-jt committed:
Merge branch 'master' of https://github.com/lim-jt/ip

rxchell commented on src/main/java/main/Parser.java:
```diff
@@ -0,0 +1,135 @@
+     * @return The EventCommand object corresponding to the parsed input.
+     * @throws DashException If the input is missing parts or has an incorrect format.
+     */
+    private static Command parseEventCommand(String input) throws DashException {
```
Good handling of the different exceptions

rxchell commented on src/main/java/exception/EmptyDescriptionException.java:
```diff
@@ -0,0 +1,16 @@
+     *
+     * @param message The detail message, which is saved for later retrieval by the {@link #getMessage()} method.
+     */
+    public EmptyDescriptionException(String message) {
```
What's the purpose of the different exceptions when all of them has the same method as Dash Exception?

rxchell commented on src/main/java/main/Storage.java:
```diff
@@ -0,0 +1,185 @@
+            appendToFile(tasks.get(i).toFile());
+        }
+        Ui.insertLine();
+        System.out.println("Nice! I've marked this task as done:");
```
Is this for the chatbot in JavaFX GUI or the terminal?

rxchell commented on src/main/java/main/Ui.java:
```diff
@@ -0,0 +1,195 @@
+            "| '--------------' || '--------------' || '--------------' || '--------------' |\n" +
+            " '----------------'  '----------------'  '----------------'  '----------------'";
+    private static final String HORIZONTAL_LINE = "________________________________________";
+    private static final String GOODBYE = "   ___                _ _                  _ \n" +
```
Good use of static

rxchell commented on src/main/java/main/Ui.java:
```diff
@@ -0,0 +1,195 @@
+     */
+    public static void printEvent(TaskList tasks) {
+        int num = tasks.size();
+        Ui.insertLine();
```
Do you need another method for the insert line format? Since Ui.insertLine() is added at the start and end of the methods in this class.

rxchell reviewed (commented):
LGTM, just some suggestions in the code.

lim-jt committed:
Add GUI

lim-jt committed:
Merge branch 'branch-A-Level-10'

lim-jt committed:
Update README.md

lim-jt committed:
Add asserts

lim-jt committed:
Add asserts

- Added assertions in `Storage’ class to enforce preconditions and validate assumptions.
	- Ensured that critical operations such as file access and input parsing have appropriate checks in place.

lim-jt committed:
Add assertions to Storage class

Assertions have been added to the Storage class to enhance the reliability
of task loading and file operations. These assertions validate critical
assumptions, such as the successful access to data files and input
handling.

This change helps catch potential errors early, ensuring that the application
behaves as expected during file operations.

lim-jt committed:
Merge branch 'master' of https://github.com/lim-jt/ip

lim-jt committed:
Make code quality changes

- General code cleanup and improvements for maintainability and consistency.
- Added Javadoc comments for `showUnmarkTask` and `showMarkTask` methods in `Ui`.
- Removed repeated error catching in `Parser`.

lim-jt committed:
Make code quality changes

lim-jt committed:
Make code quality changes

lim-jt committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/lim-jt/ip into branch-A-CodeQuality

lim-jt committed:
Merge pull request #2 from lim-jt/branch-A-CodeQuality

Branch a code quality

lim-jt committed:
Merge branch 'master' of https://github.com/lim-jt/ip into branch-A-CodeQuality

lim-jt committed:
Merge branch 'master' of https://github.com/lim-jt/ip

lim-jt committed:
Merge branch 'branch-A-CodeQuality'

lim-jt committed:
Update README and remove unnecessary print outs

lim-jt committed:
Add friendlier syntax

lim-jt committed:
Add friendlier syntax

lim-jt committed:
Merge branch 'branch-Extension-FriendlierSyntax'

lim-jt committed:
Add user guide

lim-jt committed:
Update README.md

lim-jt committed:
Rename Ui.png

lim-jt committed:
Merge branch 'master' of https://github.com/lim-jt/ip

lim-jt committed:
Add exit sequence

lim-jt committed:
Adjust screen height and width

lim-jt committed:
Update README.md

lim-jt committed:
Merge commit 'feb5fe9fadc91b70d7ae06ed08323e8a9137206a'

# Conflicts:
#	data/dash.txt

lim-jt committed:
Edit friendlier syntax to require a space after commands with more than
one word
